predicates/predicates_test.go

```diff
@@ -0,0 +1,308 @@
+			})
+
+			It("returns false when the application has not changed", func() {
+				Expect(hasApplicationChanged(releasePlan, releasePlanDiffTarget)).To(BeFalse())
+			})
+		})
+
+		When("calling hasTargetChanged", func() {
+			It("returns true when the target has changed", func() {
+				Expect(hasTargetChanged(releasePlan, releasePlanDiffTarget)).To(BeTrue())
+			})
+
+			It("returns false when the target has not changed", func() {
+				Expect(hasTargetChanged(releasePlan, releasePlanDiffApp)).To(BeFalse())
+			})
+		})
+	})
+
+	Context("ReleasePlanAdmission functions", func() {
+		var releasePlanAdmission, releasePlanAdmissionDiffApps, releasePlanAdmissionDiffOrigin *v1alpha1.ReleasePlanAdmission
+		var instance predicate.Predicate
+
+		BeforeAll(func() {
+			releasePlanAdmission = &v1alpha1.ReleasePlanAdmission{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "releaseplanadmission",
+					Namespace: namespace,
+				},
+				Spec: v1alpha1.ReleasePlanAdmissionSpec{
+					Applications: []string{
+						applicationName,
+					},
+					Origin: namespace2,
+					Policy: "policy",
+					PipelineRef: &tektonutils.PipelineRef{
+						Resolver: "bundles",
+						Params: []tektonutils.Param{
+							{Name: "bundle", Value: "quay.io/some/bundle"},
+							{Name: "name", Value: "release-pipeline"},
+							{Name: "kind", Value: "pipeline"},
+						},
+					},
+				},
+			}
+			releasePlanAdmissionDiffApps = &v1alpha1.ReleasePlanAdmission{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "releaseplanadmission-app",
+					Namespace: namespace,
+				},
+				Spec: v1alpha1.ReleasePlanAdmissionSpec{
+					Applications: []string{
+						"diff",
+					},
+					Origin: namespace2,
+					Policy: "policy",
+					PipelineRef: &tektonutils.PipelineRef{
+						Resolver: "bundles",
+						Params: []tektonutils.Param{
+							{Name: "bundle", Value: "quay.io/some/bundle"},
+							{Name: "name", Value: "release-pipeline"},
+							{Name: "kind", Value: "pipeline"},
+						},
+					},
+				},
+			}
+			releasePlanAdmissionDiffOrigin = &v1alpha1.ReleasePlanAdmission{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "releaseplanadmission-origin",
+					Namespace: namespace,
+				},
+				Spec: v1alpha1.ReleasePlanAdmissionSpec{
+					Applications: []string{
+						applicationName,
+					},
+					Origin: "diff",
+					Policy: "policy",
+					PipelineRef: &tektonutils.PipelineRef{
+						Resolver: "bundles",
+						Params: []tektonutils.Param{
+							{Name: "bundle", Value: "quay.io/some/bundle"},
+							{Name: "name", Value: "release-pipeline"},
+							{Name: "kind", Value: "pipeline"},
+						},
+					},
+				},
+			}
+			ctx := context.Background()
+
+			Expect(k8sClient.Create(ctx, releasePlanAdmission)).Should(Succeed())
+			Expect(k8sClient.Create(ctx, releasePlanAdmissionDiffApps)).Should(Succeed())
+			Expect(k8sClient.Create(ctx, releasePlanAdmissionDiffOrigin)).Should(Succeed())
```

Comment:
Why do you create them? You don't consume them from the cluster isn't it?

Reply:
fixed